android/src/main/java/com/imagepicker/ImagePickerModule.java

```diff
@@ -167,9 +167,9 @@ public void onActivityResult(Activity activity, int requestCode, int resultCode,
 
             case REQUEST_LAUNCH_VIDEO_CAPTURE:
                 if (options.saveToPhotos) {
-                    saveToPublicDirectory(data.getData(), reactContext, "video");
+                    saveToPublicDirectory(cameraCaptureURI, reactContext, "video");
                 }
-                onVideoObtained(data.getData());
+                onVideoObtained(cameraCaptureURI);
```

Comment:
This works, but would it be better to do a null check on data, and then use cameraCaptureURI only if its null?

Reply:
The fact is data.getData() and cameraCaptureURI should be same. So no need for checking.